src/include/daos_obj.h

```diff
@@ -355,6 +355,54 @@ daos_obj_generate_id(daos_obj_id_t *oid, daos_ofeat_t ofeats,
 	oid->hi |= hdr;
 }
 
+typedef enum {
+	DAOS_NO_RDD,
+	DAOS_ANY_RDD,
+	DAOS_RP_RDD,
+	DAOS_EC_RDD,
+} daos_oclass_rdd_t;
```

Comment:
I think that it might even be difficult for a middleware to select that. I would up-level that to the container property actually, although we should still give the ability for some specific objects to have a different layout.